src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+            }
+        }
+	    
+	    if (checkBoxLanguage.isSelected()) {
+	        // submissions may be filtered by language
+	        
+	        if (submitSelectedLanguagesRadioButton.isSelected()) {
+	            if (selectedLanguagesIndexes == null || selectedLanguagesIndexes.length == 0) {
+	                FrameUtilities.showMessage(this, "No Language selected", "No Languages selected, select at least one");
+	                return;
+
+	            }
+
+	            files = ListUtilities.filterByLanguages(files, getContest(), selectedLanguageList);
+	        }
+	    }
+	    
+
+	    int count = 0;
+	    for (File file : files) {
+	        count++;
+	        xlog("Will submit #" + count + " file =" + file.getAbsolutePath());
+	    }
+
+	    if (count == 0) {
+	        showMessage("There are no CDP samples source files  under " + cdpTextField.getText());
+	        return;
+	    }
+
+	    int result = FrameUtilities.yesNoCancelDialog(this, "Submit " + files.size() + " sample submissions?",
+	            "Submit CDP submissions");
+
+	    if (result == JOptionPane.YES_OPTION) {
+	        /**
+	         * This will submit each file found
+	         */
+	        submitter.sendSubmissions(files);
+
+	        showMessage("Submitted " + files.size() + " runs.");
+	    }
+
+
+	}
+
+	/**
+	 * Check whether cdpDir and model match
+	 * @param cdpDir base dir for CDP, parent dir for config/ dir
+	 * @return null if no issues, else a warning message about a diffence between model and cdpDir
+	 */
+	private String verifyCDP(String cdpDir) {
+		
+		// TODO compare problems in CDP with problems in model 
+		
+		return null;
+	}
+
+	/**
+	 * Returns the list of filenames that end in extension
+	 * 
+	 * @param files
+	 * @param extension
+	 * @return
+	 */
+	public static List<File> filterSource(List<File> files, String extension) {
+
+		List<File> list = new ArrayList<File>();
+		for (File file : files) {
+			if (file.getName().endsWith(extension)) {
+				list.add(file);
+			}
+		}
+		return list;
+	}
+
+	@Override
+	public String getPluginTitle() {
+		return "Submitter Pane";
+	}
+
+	@Override
+	public void setContestAndController(IInternalContest inContest, IInternalController inController) {
+		super.setContestAndController(inContest, inController);
+
+		resetFields(false);
+
+		String cdpPath = getContest().getContestInformation().getJudgeCDPBasePath();
+		String clientPath = getClientCDPPath();
+
+		if (clientPath != null) {
+			cdpPath = clientPath;
+		}
+
+		xlog("CDP dir is now at " + cdpPath);
+		cdpTextField.setText(cdpPath);
+
+		showMessage("");
+
+		submitter.setContestAndController(inContest, inController);
+		
+		
+		log = inController.getLog();
+
+	}
+
+	public void showMessage(final String message) {
+		SwingUtilities.invokeLater(new Runnable() {
+			public void run() {
+				xlog(message);
+				messageLabel.setText(message);
+				messageLabel.setToolTipText(message);
+			}
+		});
+	}
+
+
+} // @jve:decl-index=0:visual-constraint="10,10"
```

Comment:
"Someday":  delete old, useless comments like this (left over from the VisualAge days?)